src/transforms/rotation.jl

```diff
@@ -0,0 +1,42 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Rotation(rot)
+
+Rotate geometry or mesh with rotation `rot` from ReferenceFrameRotations.jl. 
+
+## Examples
+
+```julia
+Rotation(EulerAngleAxis(pi/4, [1, 0, 0]))
+```
+"""
+struct Rotation{R} <: GeometricTransform
+  rot::R
+end
+
+isrevertible(::Type{<:Rotation}) = true
+
+function preprocess(transform::Rotation, object)
+  rot = transform.rot
+  R, R⁻¹ = rot, inv_rotation(rot)
```

Comment:
Their documentation is very explicit about the `inv_rotation` limitations:

https://juliaspace.github.io/ReferenceFrameRotations.jl/stable/man/inv_rotations

Can we stick to the suggested change where `inv(rot)` is used? I understand that `inv` works with any rotation type.

Reply:
This doc is wrong. Here is the correct doc:

https://juliaspace.github.io/ReferenceFrameRotations.jl/stable/lib/library/#ReferenceFrameRotations.inv_rotation-Tuple{DCM}

Reply:
Please stick to `inv` as this is the universal function in Julia for inverses:

```julia
julia> using ReferenceFrameRotations

julia> r1 = EulerAngleAxis(1, [1, 1, 1])
EulerAngleAxis{Int64}:
  Euler angle : 1 rad  (57.2958°)
  Euler axis  : [1, 1, 1]

julia> r2 = EulerAngles(1, 1, 1)
EulerAngles{Int64}:
  R(Z) :  1 rad  ( 57.2958°)
  R(Y) :  1 rad  ( 57.2958°)
  R(X) :  1 rad  ( 57.2958°)

julia> r3 = Quaternion(1.0, 0.0, 0.0, 0.0)
Quaternion{Float64}:
  + 1.0 + 0.0⋅i + 0.0⋅j + 0.0⋅k

julia> r4 = DCM(I)
DCM{Bool}:
 1  0  0
 0  1  0
 0  0  1

julia> inv(r1)
EulerAngleAxis{Float64}:
  Euler angle : 1.0 rad  (57.2958°)
  Euler axis  : [-1.0, -1.0, -1.0]

julia> inv(r2)
EulerAngles{Int64}:
  R(X) : -1 rad  (-57.2958°)
  R(Y) : -1 rad  (-57.2958°)
  R(Z) : -1 rad  (-57.2958°)

julia> inv(r3)
Quaternion{Float64}:
  + 1.0 - 0.0⋅i - 0.0⋅j - 0.0⋅k

julia> inv(r4)
DCM{Float64}:
 1.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  1.0
```